processUsers.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import datetime
+import random
+import requests
+import string
+import sys
+
+
+def format_response(response):
+    response_json = response.json()
+    return str(response_json.get('code')) + ' : ' + response_json.get('detail')
+
+
+def create_users(email_list, magpie_url, magpie_admin_user_name, magpie_admin_password):
+    session = requests.Session()
+    response = session.post(magpie_url + "/signin", data={'user_name': magpie_admin_user_name,
+                                                          'password': magpie_admin_password,
+                                                          'provider_name': 'ziggurat'})
+    if not response.ok:
+        print(format_response(response))
+        return []
+
+    users = []
+    for email in email_list:
+        user = {'email': email,
+                'user_name': email,
+                # generate a password made of 8 random alphanumeric characters
+                'password': "".join(random.choice(string.ascii_letters + string.digits) for _ in range(8)),
+                'result': ""}
+        users.append(user)
+        response = session.post(magpie_url + "/users", data={'user_name': user['user_name'],
+                                                             'email': user['email'],
+                                                             'password': user['password'],
+                                                             'group_name': 'users'})
+        if not response.ok:
+            user['result'] = format_response(response)
+
+    # test each successful users with a login
+    for user in users:
+        if user['result'] == "":
+            session = requests.Session()
+            response = session.post(magpie_url + "/signin",
+                                    data={'user_name': user['user_name'], 'password': user['password'],
+                                          'provider_name': 'ziggurat'})
+            user['result'] = format_response(response)
+    return users
+
+
+def delete_users(user_names, magpie_url, magpie_admin_user_name, magpie_admin_password):
+    session = requests.Session()
+    response = session.post(magpie_url + "/signin", data={'user_name': magpie_admin_user_name,
+                                                          'password': magpie_admin_password,
+                                                          'provider_name': 'ziggurat'})
+    if not response.ok:
+        print(format_response(response))
+        return
+
+    users = []
+    for user in user_names:
+        response = session.delete(magpie_url + "/users/" + user)
+        users.append({'user_name': user, 'result': format_response(response)})
+    return users
+
+
+if __name__ == '__main__':
+    email_list = sys.argv[1:]
+    magpie_url = "http://localhost/magpie"
+    magpie_admin_user_name = "admin"
+    magpie_admin_password = "admin"

```

Comment:
These parameters should all be passed as input to the script.
Check `argparse` module to make a clean CLI.